src/board.js

```diff
@@ -98,6 +99,64 @@ class Board {
       }
     }
   }
+
+  detectWhenKingIsInCheck() {    
+    let whiteKingPosition, blackKingPosition;
+
+    // Znalezienie pozycji królów
+    this.forEachSquare((row, column) => {
+      const piece = this.getSquare(row, column).piece;      
+      if (piece?.name === 'king') {
+        if (piece.side === 'white') {
+          whiteKingPosition = [row, column];          
+        } else {
+          blackKingPosition = [row, column];
+        }
+      }
+    });
+
+    // Usunięcie czerwonego pola po szachu króla
+    if (this.oldWhiteKingPosition) {
+      this.getSquare(this.oldWhiteKingPosition[0], this.oldWhiteKingPosition[1]).removeCheckHighlight();
+    }
+    if (this.oldBlackKingPosition) {
+      this.getSquare(this.oldBlackKingPosition[0], this.oldBlackKingPosition[1]).removeCheckHighlight();
+    }
+
+    // Zaktualizowanie pozycji królów
+    this.oldWhiteKingPosition = whiteKingPosition;
+    this.oldBlackKingPosition = blackKingPosition;
+
+    // Sprawdzenie, czy któryś z królów jest w szachu
+    let whiteKingInCheck = false;
+    let blackKingInCheck = false;
+    
+    this.forEachSquare((row, column) => {
+      const piece = this.getSquare(row, column).piece;
+      if(piece) {
+        const futureMoves = piece.findLegalMoves(this);
+        for (const [futureMoveRow, futureMoveColumn] of futureMoves) {
+          if (futureMoveRow === whiteKingPosition[0] && futureMoveColumn === whiteKingPosition[1]) {
+            whiteKingInCheck = true;
+          }
+          if (futureMoveRow === blackKingPosition[0] && futureMoveColumn === blackKingPosition[1]) {
+            blackKingInCheck = true;
+          }
+        }
+      } 
+    });
+
+    // Oznaczenie królow w szachu
+    if (whiteKingInCheck) {
+      alert("Biały król jest w szachu!");
+      this.getSquare(whiteKingPosition[0], whiteKingPosition[1]).checkHighlight();
+    }
+
+    if (blackKingInCheck) {
+      alert("Czarny król jest w szachu!");
+      this.getSquare(blackKingPosition[0], blackKingPosition[1]).checkHighlight();
+    } 
```

Comment:
Tak jak wyżej to lepiej by było zrobić po za tą metodą, będą sytuacje w których chcemy sprawdzić czy król jest w szachu bez podświetlania pola z królem np.: przed ruchem bierki żeby nie pozwolić na ruch który odsłoni naszego króla